.github/workflows/linter.yml

```diff
@@ -10,6 +10,8 @@ on:
       - main
       - v[0-9]+-stable
 
+permissions: read-all
```

Comment:
I guess in the future it would be good to set `statuses: write` here, since we pass the token to super-linter so it can split the results to per-language statuses (see the note in https://github.com/github/super-linter/blob/main/README.md#how-to-use). But since we use only the shell linter, it doesn't matter.

Reply:
To judge from https://github.com/evverx/systemd/actions/runs/1456884989 (where I tested labeler without the write permissions) GHActions tend to complain loudly when they break due to permissions so I think hopefully we won't miss that and change the action accordingly :-)

Reply:
@mrc0mmand once the PR is merged just in case I'll try to send a follow-up PR triggering a linter warning to make sure the permissions aren't too restrictive. But I'm not sure what that actions does. Is that basically `shellcheck` on steroids? If you know how to reliable trigger warnings there it would be great if you could post it here so that I could borrow it.

Reply:
I found https://github.com/systemd/systemd/pull/20884. I'll try to revert one of those commits

Reply:
On second thought, it seems it isn't necessary after all because judging by all the PRs I looked at where the action failed it doesn't need the write permissions when only one language is used. I should have read your first comment more carefully. Sorry about that

Reply:
Looking at one particular case (https://github.com/systemd/systemd/runs/4191993794?check_suite_focus=true) it looks like the multi status never worked as it should, because the `GITHUB_TOKEN` didn't have appropriate privileges:

```
2021-11-12 19:30:46 [INFO]   The script has completed
2021-11-12 19:30:46 [INFO]   ----------------------------------------------
2021-11-12 19:30:46 [INFO]   ----------------------------------------------
2021-11-12 19:30:46 [ERROR]   ERRORS FOUND in BASH:[1]
2021-11-12 19:30:47 [INFO]   ERROR! Failed to call GitHub Status API!
2021-11-12 19:30:47 [INFO]   ERROR:[]
2021-11-12 19:30:47 [FATAL]   Exiting with errors found!
```

but looking at the super-linter's source code, it doesn't fail the script, just prints an error:

https://github.com/github/super-linter/blob/fb8212608290c165785a9f4d6e90bad904d93690/lib/linter.sh#L660-L670

> Is that basically `shellcheck` on steroids? 

Yep, it's "just" shellcheck.

Reply:
Since it seems GITHUB_SECRET isn't actually needed there I wonder if it would make sense to just not pass GITHUB_SECRET

Reply:
Yeah, I guess we can safely drop it.

Reply:
I'll try to figure out whether the status check is more pleasant to look at than the output of the whole action and depending on the results either drop the token or let it update statuses. It might take a while though